### PR TITLE
Extract operation resolvers and execute them after path helpers

### PR DESCRIPTION
**Pull request by yoichi**
Solve https://github.com/marshmallow-code/apispec/issues/136
NOTE: This fix introduce incompatible change to extension mechanism.

Thanks,

### CODE REVIEW

**Comment by sloria** [comment]
@yoichi What about this PR is backwards-incompatible? Isn't this just exposing a new hook, or am I missing something?

**Comment by yoichi** [comment]
I thought extracting a part of functionality of path_helper introduce some incompatibility. But that was  my misunderstanding.
Extensions can keep resolver functionality in path_helper, or change themselves to use new hook.

File: docs/writing_plugins.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ There are three types of helper functions:
```
Comment (sloria) [inline docs/writing_plugins.rst:12]: Above line will need to be changed to "four".

**Comment by sloria** [comment]
@yoichi OK, thanks for the clarification. If this doesn't introduce backwards incompatibilities, then I think we should go ahead and merge this.

File: docs/writing_plugins.rst [hunk 2]
```diff
@@ -8,10 +8,11 @@ Helper Functions
 
 Plugins are comprised of "helper" functions that augment the behavior of `apispec.APISpec` methods.
 
-There are three types of helper functions:
+There are four types of helper functions:
 
 * Definition helpers
 * Path helpers
+* Operation resolvers
```
Comment (sloria) [inline docs/writing_plugins.rst]: @yoichi Would it make sense to call these "Operation helpers" for consistency?

**Review by sloria — COMMENTED** [review]
Just a suggestion for a name change. Other than that, this looks good to merge.

File: docs/writing_plugins.rst (hunk 2 above)
Comment (yoichi) [reply docs/writing_plugins.rst]: Yes. They don't have to resolve something and call them helpers like existing ones seems better. I'll push a change.
Thanks for the suggestion

**Comment by sloria** [comment]
Looks good to me. Thanks!

**Pull request merged**